sonic-chassisd/scripts/chassisd

```diff
@@ -559,6 +608,184 @@ class ModuleUpdater(logger.Logger):
+            try:
+                with open(PLATFORM_JSON_FILE, 'r') as f:
+                    platform_cfg = json.load(f)
+                # Extract the "dpu_reboot_timeout" if it exists
+                self.dpu_reboot_timeout = int(platform_cfg.get("dpu_reboot_timeout", DEFAULT_DPU_REBOOT_TIMEOUT))
+            except (json.JSONDecodeError, ValueError) as e:
+                self.log_error("Error parsing {}: {}".format(PLATFORM_JSON_FILE, e))
+            except Exception as e:
+                self.log_error("Unexpected error: {}".format(e))
+
+    def deinit(self):
+        """
+        Destructor of ModuleUpdater
+        :return:
+        """
+        # Delete all the information from DB and then exit
+        for module_index in range(0, self.num_modules):
+            name = try_get(self.chassis.get_module(module_index).get_name)
+            self.module_table._del(name)
+            if self.midplane_table.get(name) is not None:
+                self.midplane_table._del(name)
+
+        if self.chassis_table is not None:
+            self.chassis_table._del(CHASSIS_INFO_KEY_TEMPLATE.format(1))
+
+    def module_db_update(self):
+        notOnlineModules = []
+        for module_index in range(0, self.num_modules):
+            module_info_dict = self._get_module_info(module_index)
+            if module_info_dict is not None:
+                key = module_info_dict[CHASSIS_MODULE_INFO_NAME_FIELD]
+
+                if not key.startswith(ModuleBase.MODULE_TYPE_DPU):
+                    self.log_error("Incorrect module-name {}. Should start with {} ".format(key,
+                                                            ModuleBase.MODULE_TYPE_DPU))
+                    continue
+
+                fvs = swsscommon.FieldValuePairs([(CHASSIS_MODULE_INFO_DESC_FIELD, module_info_dict[CHASSIS_MODULE_INFO_DESC_FIELD]),
+                                                  (CHASSIS_MODULE_INFO_SLOT_FIELD,
+                                                   module_info_dict[CHASSIS_MODULE_INFO_SLOT_FIELD]),
+                                                  (CHASSIS_MODULE_INFO_OPERSTATUS_FIELD, module_info_dict[CHASSIS_MODULE_INFO_OPERSTATUS_FIELD]),
+                                                  (CHASSIS_MODULE_INFO_SERIAL_FIELD, module_info_dict[CHASSIS_MODULE_INFO_SERIAL_FIELD])])
+                current_status = self.get_module_current_status(key)
+                if self.prev_status[module_index] == ModuleBase.MODULE_STATUS_OFFLINE:
+                    if current_status != ModuleBase.MODULE_STATUS_OFFLINE:
+                        # DPU module has been turned on, update the reboot-cause
+                        # If the reboot-cause file not already present save it
+                        # If the STATE_DB REBOOT_CAUSE table already doesn't have
+                        # an entry for the recent reboot update the DB
+                        reboot_cause = try_get(self.chassis.get_module(module_index).get_reboot_cause)
```

Comment:
Where is the reboot cause populated in the CHASSIS_DB? I don't see the implementation for this. The reboot cause is read into a local variable and that is it. Nothing else is done with the variable